core/arch/interp.c

```diff
@@ -2852,6 +2852,13 @@ client_process_bb(dcontext_t *dcontext, build_bb_t *bb)
         if (!instr_opcode_valid(inst))
             continue;
 
+# ifdef AARCH64
+        if (inst != instrlist_last(bb->ilist)) {
+            CLIENT_ASSERT(instr_get_opcode(inst) != OP_isb,
```

Comment:
This is too early in the loop body here: it needs to be below the instr_is_meta check below, as a client can insert whatever non-app-instr (i.e., "meta" or "tool") code it wants (we won't mangle it; up to client to get correctness right; might have its own reasons for using it and we shouldn't get in the way).  See how the other checks like instr_is_syscall, client_check_syscall, instr_will_be_exit_cti are down below.